tests/unit/utils/test_utils_helpers.py

```diff
@@ -20,7 +20,140 @@
 # granted to it by virtue of its status as an Intergovernmental Organization
 # or submit itself to any jurisdiction.
 
-from inspirehep.dojson.utils import force_force_list
+from __future__ import absolute_import, division, print_function
+
+import os
+import pkg_resources
+import tempfile
+
+import httpretty
+import pytest
+
+from inspirehep.utils.helpers import (
+    download_file,
+    download_file_to_record,
+    get_json_for_plots,
+    force_force_list,
+)
+
+
+@pytest.fixture
+def foo_bar_baz():
+    return pkg_resources.resource_string(
+        __name__, os.path.join('fixtures', 'foo-bar-baz'))
+
+
+class StubRecord(object):
+    def __init__(self, files):
+        self.files = files
+
+
+@pytest.mark.httpretty
+def test_download_file(foo_bar_baz):
+    httpretty.register_uri(
+        httpretty.GET, 'http://example.com/foo-bar-baz', body=foo_bar_baz)
+
+    filename = next(tempfile._get_candidate_names())
+
+    expected = filename
+    result = download_file(
+        'http://example.com/foo-bar-baz', output_file=filename)
+
+    assert expected == result
+
+    with open(filename, 'rb') as f:
+        assert 'foo\nbar\nbaz\n' == f.read()
+
+    os.remove(filename)
+
+
+@pytest.mark.httpretty
+def test_download_file_does_not_create_a_file_if_the_request_fails():
+    httpretty.register_uri(
+        httpretty.GET, 'http://example.com/500', body='', status=500)
+
+    filename = next(tempfile._get_candidate_names())
+
+    expected = filename
+    result = download_file(
+        'http://example.com/500', output_file=filename)
+
+    assert expected == result
+
+    with pytest.raises(IOError):
+        open(filename, 'rb')
+
+
+@pytest.mark.httpretty
+def test_download_file_raises_if_called_without_output_file():
+    httpretty.register_uri(
+        httpretty.GET, 'http://example.com/foo-bar-baz', body=foo_bar_baz)
+
+    with pytest.raises(IOError):
+        download_file('http://example.com/foo-bar-baz')
```

Comment:
This instead documents the weird `None` default argument, which I don't think it's a good idea either.